### PR TITLE
Resolve 'Implement RF communication for Teensy Telemetry '

### PR DESCRIPTION
**Pull request by edproudlove**

### CODE REVIEW

File: Teensy/src/rfModule/rfModule.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+
+    return true;
+}
+
+
+bool rfModule_sendline(sensorData_s *pSensorData)
+{
+    //write the data using pointers to sensor data + checksum + newline
+    radio.write(&pSensorData->wheelSpeed_mph.fr, sizeof(pSensorData->wheelSpeed_mph.fr));
```
Comment (MellingJJ) [inline Teensy/src/rfModule/rfModule.cpp:58]: To make it easier to parse the data on the other side we want to split all the data with commas. 

I would say the best way to handle this is to create a static 32 byte char array to you can use something like snprintf to create your formatted data chunk before calling radio.write on it. I don't know exactly how .write function works but I would have thought there shouldn't be any issue with using the same array for each call

**Review by MellingJJ — CHANGES_REQUESTED** [review]
@edproudlove A couple of points to start with :)

File: Teensy/src/rfModule/rfModule.cpp (hunk 1 above)
Comment (edproudlove) [reply Teensy/src/rfModule/rfModule.cpp:58]: Ok ill take a look

File: Teensy/src/tli/tli.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    //Send in whole 32 byte chunks using the usedBufferSpace too calc how many chunks that is
+    for (int i = 0; i < (floor(usedBufferSpace / 32)); i++ ){
+        bufferCounter = 32*i;
+        bufferPtr = buffer + bufferCounter;
+        radio.write(bufferPtr, 32); 
+    }
+    
+    //Send anything remaining after the 32 byte chunks
+    if ((usedBufferSpace % 32) != 0) {
```
Comment (MellingJJ) [inline Teensy/src/tli/tli.cpp:86]:
```suggestion
radio.write(pTxBuffer + bufferIdx + 32, txBufferSize % 32)
```
We should be fine with something like this instead. I would be surprised if radio.write() didn't handle zero-sized buffers efficiently

File: Teensy/src/tli/tli.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    pSensorData->batteryVoltage_dV, 
+    pSensorData->throttlePos_mm, 
+    pSensorData->fuelPressure_pa, 
+    checksum);
+
+    //Send in whole 32 byte chunks using the usedBufferSpace too calc how many chunks that is
+    for (int i = 0; i < (floor(usedBufferSpace / 32)); i++ ){
+        bufferCounter = 32*i;
+        bufferPtr = buffer + bufferCounter;
+        radio.write(bufferPtr, 32);
```
Comment (MellingJJ) [inline Teensy/src/tli/tli.cpp:82]:
```suggestion
        radio.write(pTxBuffer + (32 * i), 32); 
```

**Review by MellingJJ — CHANGES_REQUESTED** [review]
@edproudlove Looks good, better method than what I initially had in mind. Made some comments on name/type improvements and little things. Back to you

**Comment by edproudlove** [comment]
@MellingJJ That should be most of the changes from the review. I don't think that bufferPtr or bufferIdx are needed anymore and I declared int i outside the for loop to handle anything after the 32 byte packets.

File: Teensy/src/tli/tli.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    pSensorData->steeringWheelPos_degrees, 
+    pSensorData->gyro.x, 
+    pSensorData->gyro.y, 
+    pSensorData->gyro.z, 
+    pSensorData->batteryVoltage_dV, 
+    pSensorData->throttlePos_mm, 
+    pSensorData->fuelPressure_pa, 
+    checksum);
+
+    //Send i whole 32 byte chunks using the usedBufferSpace too calc i (could be a while loop)
```
Comment (MellingJJ) [inline Teensy/src/tli/tli.cpp:77]:
```suggestion
    //Send i whole 32 byte chunks using the usedBufferSpace too calc i (could be a while loop)
    int32_t i;
```
I think its a little neater if  `i` is declared down here

**Review by MellingJJ — CHANGES_REQUESTED** [review]
@edproudlove Yeah, that looks good. Just that one tiny thing. 

I would like to do some testing first before merging this in, have you been able to get in contact with Cameron to see if you can get some HW?

**Comment by edproudlove** [comment]
@MellingJJ  I asked about a week ago and the rf modules hadn't come yet, so ill give him another message.

**Pull request merged**